### PR TITLE
Rename DateTime to Instant and add new date and time types

### PR DESCRIPTION
**Pull request by ulrikrasmussen**
This implements the necessary changes for compatibility with !1827 on GitLab.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -10,6 +10,8 @@
 * Added `unitName` field to `GuardError` to indicate which file a guard error occurred in when using multi-file projects.
 * Added a terminate command, which disallows applying further events on a contract.
 * Added support for `contract entrypoint`.
+* Rename the DateTime CSL type to Instant.
+* Add support for new date/time types.
```
Comment (jespera) [inline Readme.md:14]: These two should probably have in in a new `### Unrelease` section above?

File: Readme.md [hunk 2]
```diff
@@ -10,6 +10,8 @@
 * Added `unitName` field to `GuardError` to indicate which file a guard error occurred in when using multi-file projects.
 * Added a terminate command, which disallows applying further events on a contract.
 * Added support for `contract entrypoint`.
+* Rename the DateTime CSL type to Instant.
```
Comment (bjornbugge) [inline Readme.md:13]:
```suggestion
* Rename the `DateTime` CSL type to Instant.
```

File: Readme.md (hunk 1 above)
Comment (bjornbugge) [inline Readme.md:14]:
```suggestion
* Added support for new date/time types.
```

Comment (ulrikrasmussen) [reply Readme.md]: Yes, good catch. The automatic merge git is doing unfortunately always ends up doing this when there has been a release after the branch was created.

**Review by jespera — APPROVED** [review]

**Pull request merged**